### PR TITLE
Update Scrivito Definitions to match Scrivito version 1.22

### PR DESCRIPTION
**Pull request by juliankrieger**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://www.scrivito.com/scrivito-js-1-22-0-released-featuring-language-aware-search-bc64a95b4cb84800>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/scrivito/index.d.ts [hunk 1]
```diff
@@ -49,8 +49,8 @@ interface OptimizeDefinition {
 
 export class Binary {
     private constructor();
-    static upload(source: Blob | File, options: { filename: string, contentType?: string }): FutureBinary;
-    static uplload(source: File, options?: { filename?: string, contentType?: string }): FutureBinary;
+    static upload(source: Blob | File, options: { filename: string; contentType?: string }): FutureBinary;
```
Comment (eric-the-viking) [inline types/scrivito/index.d.ts:52]: Can you create a new interface for the upload options?

File: types/scrivito/index.d.ts [hunk 2]
```diff
@@ -49,8 +49,8 @@ interface OptimizeDefinition {
...
+    static uplload(source: File, options?: { filename?: string; contentType?: string }): FutureBinary;
```
Comment (eric-the-viking) [inline types/scrivito/index.d.ts:53]: Is this a typo?

File: types/scrivito/index.d.ts [hunk 3]
```diff
@@ -165,10 +166,10 @@ interface ConfigOptions {
     // Hard to type
     constraintsValidation?: (constraints: any) => any;
     endpoint?: string;
-    priority?: 'foreground' | 'background';
+    priority?: "foreground" | "background";
```
Comment (eric-the-viking) [inline types/scrivito/index.d.ts:169]: Please create a type for the priority options

File: types/scrivito/index.d.ts [hunk 4]
```diff
@@ -521,27 +579,36 @@ export function provideAuthGroups(options: AuthGroupsOptions): void;
...
 export function provideObjClass(name: string, contentClassOrOptions: ObjClassOptions | AbstractObjClass): ObjClass;
-export function provideWidgetClass(name: string, contentClassOrOptions: WidgetClassOptions | AbstractWidgetClass): WidgetClass;
+export function provideWidgetClass(
+    name: string,
+    contentClassOrOptions: WidgetClassOptions | AbstractWidgetClass,
+): WidgetClass;
 
 export function registerComponent(name: string, component: WidgetComponent | ObjComponent): void;
-export function renderPage(page: any, renderFunction: () => void): Promise<{ result: Obj, preloadDump: any }>;
+export function renderPage(page: any, renderFunction: () => void): Promise<{ result: Obj; preloadDump: any }>;
```
Comment (eric-the-viking) [inline types/scrivito/index.d.ts:586]: `page` is a Scrivito.Obj

**Review by eric-the-viking — CHANGES_REQUESTED** [review]
Some minor adjustments

File: types/scrivito/index.d.ts (hunk 2 above)
Comment (juliankrieger) [reply types/scrivito/index.d.ts:53]: yes! thanks :)

File: types/scrivito/index.d.ts (hunk 1 above)
Comment (juliankrieger) [reply types/scrivito/index.d.ts:52]: done

File: types/scrivito/index.d.ts (hunk 4 above)
Comment (juliankrieger) [reply types/scrivito/index.d.ts:586]: done

**Review by eric-the-viking — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -20,7 +20,7 @@
         "update-codeowners": "node scripts/update-codeowners.js",
         "test-all": "node node_modules/@definitelytyped/dtslint-runner/dist/index.js --path .",
         "test": "dtslint types",
-        "lint": "dtslint types",
+        "lint": "dtslint types/scrivito",
```
Comment (sandersn) [inline package.json:23]: please revert this change

`dtslint` has a quirky/undiscoverable interface: it accepts `types/scrivito` or `types scrivito`. The latter lets you write `npm test scrivito`

**Review by sandersn — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (juliankrieger) [reply package.json:23]: Ah, I didn't see how this was supposed to work. Sorry for pushing my changes

**Review by sandersn — APPROVED** [review]
